datadog/dogstatsd/base.py

```diff
@@ -826,6 +828,13 @@ def flush_aggregated_metrics(self):
         for m in metrics:
             self._report(m.name, m.metric_type, m.value, m.tags, m.rate, m.timestamp)
 
+        buffered_metrics = self.aggregator.flush_aggregated_buffered_metrics()
+        send_method = self._send
+        self._send = self._send_to_buffer
```

Comment:
Why do we need to change the `_send` method when flushing aggregated metrics? 

The way I read this seems like we are buffering twice ex: histogram/distribution/timing in the buffered metric class, and then again by writing to the sender buffer via `_send_to_buffer`. 

Also - with aggregation enabled, `_start_flush_thread` is given the `flush_aggregated_metrics` method, so in this case will `flush_buffered_metrics` ever be called by the flush thread? 

Please correct me if I am wrong/misunderstanding.